python/meep.i

```diff
@@ -81,6 +84,78 @@ static double py_callback_wrap(const meep::vec &v) {
   $1 = PyCallable_Check($input);
 }
 
+// Typemap suite for vector3
+
+%typemap(in) vector3 {
+    if(!pyv3_to_v3($input, &$1)) {
+        SWIG_fail;
+    }
+}
+
+// Typemap suite for GEOMETRIC_OBJECT
+
+%typecheck(SWIG_TYPECHECK_POINTER) GEOMETRIC_OBJECT {
+    PyObject *geom_mod = PyImport_ImportModule("geom");
+    PyObject *geom_dict = PyModule_GetDict(geom_mod);
+    PyObject *py_sphere = PyDict_GetItemString(geom_dict, "GeometricObject");
+    $1 = PyObject_IsInstance($input, py_sphere);
+    Py_XDECREF(geom_mod);
```

Comment:
I wasn't decrementing the the other two objects because those are weak references.  Using your solution eliminates the need for them anyway. Fixed.